python/astra/data3d.py

```diff
@@ -52,11 +54,11 @@ def link(datatype, geometry, data):
     :returns: :class:`int` -- the ID of the constructed object.
     
     """
-    if not isinstance(data,np.ndarray):
+    if not isinstance(data,np.ndarray) and not isinstance(data,GPULink):
         raise ValueError("Input should be a numpy array")
-    if not data.dtype==np.float32:
+    if not isinstance(data,GPULink) and not data.dtype==np.float32:
```

Comment:
`and data.dtype != np.float32`
The logic here (and below) is that anything which is not a `GPULink` is a `numpy.ndarray`, which is true *currently*. Later changes will be easier if this code tests for `isinstance(data, np.ndarray)` explicitly.